### PR TITLE
Wire up order app title

### PR DESCRIPTION
**Pull request by zephraph**
Depends on https://github.com/artsy/reaction/pull/1241
Addresses [PURCHASE-433](https://artsyproduct.atlassian.net/browse/PURCHASE-433).

This will require a bump in reaction once 1241 is merged. Hooks up the SSR aspect of the title rendering.

### CODE REVIEW

File: src/desktop/apps/order2/server.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ app.get('/order2/:orderID*', async (req, res, next) => {
   try {
     const user = req.user && req.user.toJSON()
 
-    const { ServerApp, redirect, status } = await buildServerApp({
+    const { ServerApp, redirect, status, headTags } = await buildServerApp({
```
Comment (sweir27) [inline src/desktop/apps/order2/server.js]: Will we need to do this same thing for the other new-style apps? (artist, collect)

Comment (zephraph) [reply src/desktop/apps/order2/server.js]: Not unless they needed to set the title via the app.

File: src/desktop/apps/order2/server.js [hunk 2]
```diff
@@ -44,6 +44,7 @@ app.get('/order2/:orderID*', async (req, res, next) => {
         styledComponents: true,
       },
       blocks: {
+        head: () => headTags,
```
Comment (damassi) [inline src/desktop/apps/order2/server.js]: Looking through https://github.com/tizmagik/react-head this seems like a solid choice, but one thing i'm wondering about is the ability to naturally merge `headTags` into other hand-written solutions [such as this](https://github.com/artsy/force/blob/master/src/desktop/apps/artist2/components/Meta.js) which is [used here](https://github.com/artsy/force/blob/master/src/desktop/apps/artist2/server.js#L66).

I imagine it's as simple as passing `headTags` into `<Meta>` and then interpolating via `{headTags}`?

Comment (zephraph) [reply src/desktop/apps/order2/server.js]:
```jsx
{
  head: () => 
  <> 
    ${headTags}
    <Meta/>
  </>
}
```

That would probably work. I think this library would give us a bit of refactoring opportunities there though.

Comment (damassi) [reply src/desktop/apps/order2/server.js]: 👍

**Comment by sweir27** [comment]
Eep should we have waited for the reaction bump?

**Pull request merged**